### PR TITLE
fix: changed to do redirects on server instead of client side [render preview]

### LINKED ISSUE #2: Fix redirect page
## Description

When redirects are triggered, a plain white page is displayed momentarily. This affects the user experience by making the transition feel slow and unpolished. The primary logic for handling the redirect seems to take time to execute.

##Expected Behavior

The redirection process should be faster and visually appealing, ensuring a smoother user experience. Ideally:
- Reduce the time taken for the redirect logic to execute.
- Replace the white page with a simple, styled loading indicator or transition effect to enhance the aesthetic.

### PR DESCRIPTION
**Pull request by SGI-CAPP-AT2**
- before this redirects were made on client side which added extra page render
- now it'll send redirect to get request
- links shortified before this will not be affected but this change will not be available for them

closes: #2 

Results:
Right now if you receive GET request (Request sent by browser) to `links/r/:id` it'll send redirect to user so that the render of blank page will be avoided

Changes in nutshell:
image

Screenshots:
Before (Still working): when we do GET to /r/ of client it'll always serve a blank page
image

After: when we do GET to /r/ of server it'll redirect which will always show the desired web page instead of blank page
image

### CODE REVIEW

**Comment by SGI-CAPP-AT2** [comment]
@mendacium-a11y , 
If anything is going wrong in the PR please let me know

**Comment by mendacium-a11y** [comment]
can you add a video recording of the commit working locally

**Comment by SGI-CAPP-AT2** [comment]
@mendacium-a11y,

https://github.com/user-attachments/assets/5a1f46c9-c6e2-4188-b59a-f1a2dc4d724c

You can see that `[5173]/r/:id` shows some blank page and you can go to previous page
where as `[3000]/links/r/:id` directly redirects

**Comment by SGI-CAPP-AT2** [comment]
shortened urls after deploying this change might look like `shortify-ahj5.onrender.com/links/r/ididid`

File: client/src/pages/Home.tsx
Comment (mendacium-a11y) [inline client/src/pages/Home.tsx:1]: lets deprecate the existing way to redirect, and just keep the server side one, with the same path of /r/:key

**Comment by SGI-CAPP-AT2** [comment]
@mendacium-a11y done

File: client/src/pages/Home.tsx
Comment (SGI-CAPP-AT2) [reply client/src/pages/Home.tsx:1]: yes, done

**Pull request merged**